src/main/java/today/sesac/shoutify/post/dto/response/PostSummaryResponseDto.java

```diff
@@ -3,17 +3,49 @@
 import java.time.LocalDateTime;
 import lombok.AllArgsConstructor;
 import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import today.sesac.shoutify.post.entity.Post;
 
+/**
+ * 게시글 목록 조회 시 하나의 게시글을
+ * 나타내기 위한 dto.
+ */
+//image 값이 null인 경우도 있어서 image는 final 제외 -> 빌더쓰면 안 붙여도 됨
 @Getter
-@AllArgsConstructor
+@AllArgsConstructor(staticName = "of")
+@RequiredArgsConstructor(staticName = "of")
 public class PostSummaryResponseDto {
-    private Long postId;
-    private String nickname;
-    private String afterTitle;
-    private String afterContent;
-    private LocalDateTime createdAt;
-    private int reactionCount;
-    private int commentCount;
-    private String conceptType;
+
+    private final Long postId;
+
+    private final String nickname;
+
+    private final String afterTitle;
+
+    private final String afterContent;
+
+    private final LocalDateTime createdAt;
+
+    private final int reactionCount;
+
+    private final int commentCount;
+
+    private final String conceptType;
+
     private String imageUrl;
+
+    /**
+     * Service 에서 Post 객체를 Dto 객체로 바꾸기 위한 정적 메소드
+     * Concept enum 타입을 문자열로 변환.
+     *
+     * @param post          Post 객체
+     * @param reactionCount 반응 갯수(현재 0으로 고정)
+     * @param commentCount  댓글 갯수
+     * @return Page 객체 내부에 들어가는 Dto 객체
+     */
+    public static PostSummaryResponseDto of(Post post, int commentCount, int reactionCount) {
+        return new PostSummaryResponseDto(post.getId(), post.getAuthor().getNickname(),
+                post.getAfterTitle(), post.getAfterContent(), post.getCreatedAt(),
+                reactionCount, commentCount, post.getConceptType().name(), post.getImageUrl());
```

Comment:
P2
여기서 name()으로 ENUM의 이름을 주는 이유가 ENUM 상수 그대로 내려주기 위한 거 맞나요? 그런 거라면, jackson 변환 시 ENUM은 name을 기본으로 변환하기 때문에, 해당 dto의 필드를 enum으로 두는 게 타입 제한 등의 방면에서 더 일관성 있을 것 같습니다.

```suggestion
reactionCount, commentCount, post.getConceptType(), post.getImageUrl());
```

위에서 필드도 이렇게 바꾸고, 돌려보시면 응답은 동일하게 옵니다. 테스트 ㄱㄱ
```
private final Concept conceptType;
```

Reply:
오 자동으로 변환을 해주네요 그럼 굳이 dto에서 꺼내서 string으로 넘길 필요가 없어서 수정하겠습니다 클라이언트에서 enum 이름을 잘못 보낼 경우에는 .... 제가 에러처리 해야겠군요 하하

Reply:
음... 우선, 에러 처리를 어떤 걸 고민하시는지 모르겠지만, 1차적으로는 이런 식으로 spring에서 에러를 발생시키고, 그걸 잡고 있기는 합니다.

 이런 걸 고려하실 때는 포스트맨으로 직접 다양하게 보내 보면서 어떤 예외가 발생하는지 직접 확인해 보시면 좋아요~. 현재 어떤 식으로 에러를 발생시키고, 어떻게 잡고 있는지, 나는 어떤 걸 의도하는지를 고민해 보시면 좋겠습니다. 

공통 에러 처리에서 이런 걸 추가하면 좋겠다는 게 있으면 말씀해 주세요~ #96 에서 처리하겠습니다.

```shell
org.springframework.web.method.annotation.MethodArgumentTypeMismatchException: Method parameter 'concept': Failed to convert value of type 'java.lang.String' to required type 'today.sesac.versebyverse.global.domain.Concept'; Failed to convert from type [java.lang.String] to type [@org.springframework.web.bind.annotation.RequestParam today.sesac.versebyverse.global.domain.Concept] for value [DRAMAA]
```